### ISSUE TITLE
Script break in Yuan Bo's quest battle

### ISSUE THREAD

**Issue by paperpancake** [chadvandy/cbfm_wh3#1702]
In vanilla, line 123 of `battle\quest_battles\yuan_bo\battle_script.lua` is currently:

`play_sound(v(0,0),new_sound("Play_Movie_WH3_DLC24_QB_Unmute", false, false));`

The `new_sound` function does not appear to exist, which causes a script break. You can verify this with simple debug output statements before and after that line.

I haven't seen this script break have major consequences in vanilla, but with some battle mods it results in a UI lock that forces the player to Alt+F4.

I'll propose a fix in a comment below.

**Comment by paperpancake** [comment]
Possibly the most straightforward fix is to create the global `new_sound` function that the line is trying to call. I think `new_sfx` is likely what it should be calling instead, so this appears sufficient:

```
function new_sound(soundfile, _is_cinematic, _is_vo)
        return new_sfx(soundfile, _is_cinematic, _is_vo);
end;
```

**Comment by paperpancake** [comment]
Or, if you want to take extra care to avoid adding a global function except where necessary, it could be something like:

```
local function is_yuan_bo_dragon_fang_quest()
    return bm:is_quest_battle() and bm:player_army_is_faction("wh3_dlc24_cth_the_celestial_court") and bm:get_first_non_player_army():faction_key() == "wh3_main_cth_cathay_qb1";
end;

if is_yuan_bo_dragon_fang_quest() then

    function new_sound(soundfile, _is_cinematic, _is_vo)
        return new_sfx(soundfile, _is_cinematic, _is_vo);
    end;

end;
```

**Comment by Drazhir2** [comment]
Thanks for the detailed report, It is indeed an issue, you can check which quest battle it is with cco.

Here is the fix [zzz_cbfm_yuan_bo_qb.txt](https://github.com/chadvandy/cbfm_wh3/files/15214909/zzz_cbfm_yuan_bo_qb.txt) (rename to .lua and drop in script/battle/mod)

**Comment by Drazhir2** [comment]
CA broke it in 5.0 then fixed it in 5.1
![image](https://github.com/user-attachments/assets/04382cbe-618c-4a8f-8779-58ae38b263dd)
